misc/utils.cpp

```diff
@@ -0,0 +1,131 @@
+#include "utils.h"
+
+#include <fstream>
+#include <memory>
+#include <regex>
+#include <algorithm>
+#include <stdexcept>
+#include <iterator>
+
+#include <google/protobuf/util/json_util.h>
+#include <google/protobuf/util/type_resolver_util.h>
+
+using namespace std;
+using namespace google::protobuf;
+
+namespace dash{
+
+static const string GLOBAL_TYPE_URL_PREFIX = "sonic";
+
+// According to the definition, TypeResolver is thread-safe
+static unique_ptr<util::TypeResolver> resolver(
+    util::NewTypeResolverForDescriptorPool(
+        GLOBAL_TYPE_URL_PREFIX,
+        DescriptorPool::generated_pool()));
+
+// Table name: DASH_ROUTE_RULE_TABLE
+// =>
+// Type url: sonic/dash.route_rule.RouteRule
+string TableNameToTypeUrl(const string &table_name) {
+    static regex rgx("^([^_]*)_(.+)_TABLE$");
```

Comment:
nit: the performance of regex is usually not that ideal. maybe simply using find and string_view will be a better choice.

Reply:
IMHO, I feel regex is better at this situation.
1. This pattern object is static and therefore will only be compiled to NFA once. Without backtracking and compilation time, regex NFA is not much worse than linear algorithm.
2. At this point, I feel regex is cleaner and easier to extend. Because we can directly change the pattern if we want to add more information.